### PR TITLE
feat(edge,registry): pull and store in memory private registry credentials (EE-2567)

### PR DESCRIPTION
**Pull request by hookenz**
Implements private edge registries in edge stacks (agent side) [EE-2567](https://portainer.atlassian.net/browse/EE-2567)

### CODE REVIEW

File: cmd/docker-credential-portainer/portainer_helper.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		log.Fatalf("error opening file: %v", err)
+	}
+	defer f.Close()
+	log.SetOutput(f)
+
+	if serverURL == "" {
+		return "", "", credentials.NewErrCredentialsMissingServerURL()
+	}
+
+	log.Printf("GET ServerURL=%s", serverURL)
```
Comment (prabhat83) [inline cmd/docker-credential-portainer/portainer_helper.go:35]: maybe change this message to `portainer server URL`

File: edge/registry/server.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	return h
+}
+
+func (handler *Handler) LookupHandler(rw http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	stackManager := handler.EdgeManager.GetStackManager()
+	if stackManager == nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve stack manager", errors.New("Stack manager is not available")}
+	}
+
+	serverUrl, _ := request.RetrieveQueryParameter(r, "serverurl", false)
```
Comment (prabhat83) [inline edge/registry/server.go:42]: should we skip running below code if serverUrl is ""

File: cmd/docker-credential-portainer/portainer_helper.go (hunk 1 above)
Comment (hookenz) [reply cmd/docker-credential-portainer/portainer_helper.go:35]: It's not the Portainer Server URL,  it's actually more like the registry url.  Docker calls it a server URL.

File: edge/registry/server.go (hunk 2 above)
Comment (hookenz) [reply edge/registry/server.go:42]: Possibly, but the response needs to be empty credentials.  If not, you may not be able to deploy a public dockerhub stack.

**Review by prabhat83 — APPROVED** [review]

File: dev-scripts/build.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@ function build() {
     docker rmi -f "$1" &>/dev/null || true
 
     msg "Image build..."
-    docker build --no-cache -t "$1" -f build/linux/Dockerfile . &>/dev/null
+    docker build --no-cache -t "$1" -f build/linux/alpine.Dockerfile . &>/dev/null
```
Comment (hookenz) [inline dev-scripts/build.sh]: This is wrong!

**Pull request merged**